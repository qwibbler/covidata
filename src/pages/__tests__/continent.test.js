import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import Continent from '../continent';

const mockStore = configureMockStore();
const store = mockStore({
  home: {
    countries: [
      { name: 'United States', href: 'http://US/' },
      { name: 'India', href: 'http://IN/' },
      { name: 'Pakistan', href: 'http://PK/' },
    ],
    data: {
      US: {},
      Pakistan: {},
    },
  },
});

it('renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <Continent randOpacity={() => ({ opacity: '0.25' })} />
      </Router>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
