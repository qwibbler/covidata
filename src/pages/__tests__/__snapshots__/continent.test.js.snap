// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="countries"
>
  <h1 />
  <div
    className="countries tabs"
  >
    <a
      className=" tab"
      href="#/continent/undefined/undefined/US/US/"
      onClick={[Function]}
    >
      <div
        className="country tab"
      >
        <div
          className="div-bg"
          style={
            Object {
              "opacity": NaN,
            }
          }
        />
        <h2>
          US
        </h2>
        <div
          className="info block"
        >
          <h3>
            310232863
          </h3>
          <p>
            Population
          </p>
          <h3 />
          <p>
            Confirmed
          </p>
          <h3>
            NaN
            %
          </h3>
          <p>
            Percent
          </p>
        </div>
      </div>
    </a>
    <a
      className=" tab"
      href="#/continent/undefined/undefined/Pakistan/PK/"
      onClick={[Function]}
    >
      <div
        className="country tab"
      >
        <div
          className="div-bg"
          style={
            Object {
              "opacity": NaN,
            }
          }
        />
        <h2>
          Pakistan
        </h2>
        <div
          className="info block"
        >
          <h3>
            184404791
          </h3>
          <p>
            Population
          </p>
          <h3 />
          <p>
            Confirmed
          </p>
          <h3>
            NaN
            %
          </h3>
          <p>
            Percent
          </p>
        </div>
      </div>
    </a>
  </div>
</section>
`;
